packages/components/src/date-time/date.js

```diff
@@ -1,24 +1,132 @@
 /**
  * External dependencies
  */
-import ReactDatePicker from 'react-datepicker';
+import classnames from 'classnames';
 import moment from 'moment';
+import { DayPickerSingleDateController } from 'react-dates';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../button';
 
 /**
  * Module Constants
  */
 const TIMEZONELESS_FORMAT = 'YYYY-MM-DDTHH:mm:ss';
 
-function DatePicker( { currentDate, onChange, ...args } ) {
-	const momentDate = currentDate ? moment( currentDate ) : moment();
-	const onChangeMoment = ( newDate ) => onChange( newDate.format( TIMEZONELESS_FORMAT ) );
+class DatePicker extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.state = {
+			descriptionVisible: false,
+			// This is used to create a unique ID for the datepicker help overlay.
+			// See: https://github.com/WordPress/gutenberg/pull/7621/files#r217687704
+			uniqueId: Math.random().toString().substr( 2 ),
+		};
+
+		this.onClickDescriptionToggle = this.onClickDescriptionToggle.bind( this );
+	}
+
+	onClickDescriptionToggle() {
+		this.setState( { descriptionVisible: ! this.state.descriptionVisible } );
+	}
+
+	render() {
+		const { currentDate, onChange } = this.props;
+
+		const momentDate = currentDate ? moment( currentDate ) : moment();
+		const momentTime = {
+			seconds: momentDate.seconds(), minutes: momentDate.minutes(),
+			hours: momentDate.hours(),
+		};
+
+		const onChangeMoment = ( newDate ) => {
```

Comment:
We're inconsistent in how we create callback functions, with `onClickDescriptionToggle` bound to the instance and this one defined inline. We should be consistent one way or the other, preferably the former ([reference](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md)).